app/javascript/types/account.ts

```diff
@@ -1,27 +1,31 @@
 import type { Record } from 'immutable';
 
-type CustomEmoji = Record<{
-  shortcode: string;
-  static_url: string;
-  url: string;
-}>;
+import type { CustomEmoji, CustomEmojiRawValues } from './custom_emoji';
 
-type AccountField = Record<{
+interface AccountFieldRawValues {
   name: string;
   value: string;
   verified_at: string | null;
-}>;
+}
+type AccountField = Record<AccountFieldRawValues>;
+
+interface AccountRoleRawValues {
+  color: string;
+  id: string;
+  name: string;
+}
+type AccountRole = Record<AccountRoleRawValues>;
 
-interface AccountApiResponseValues {
+interface AccountApiRawValues {
   acct: string;
   avatar: string;
   avatar_static: string;
   bot: boolean;
   created_at: string;
   discoverable: boolean;
   display_name: string;
-  emojis: CustomEmoji[];
-  fields: AccountField[];
+  emojis: CustomEmojiRawValues[];
```

Comment:
Please check if those are really arrays, or `List<CustomEmojiRawValues>`.

If they are stored in the state, then they are most probably an Immutable structure and not a plain object.

(same for every other arrays in those types)

Reply:
These are correct.
All RawValues are explain raw values of api response.

Your this comment is right for NormalizedAccountValues.
I will fix NormalizedAccountValues.

Reply:
I dont think it is normal that those fields are arrays and not immutable structures.

Everything in the state needs to be an Immutable structure, otherwise you will encounter bugs. For example, if `emojis` is a plain array and an item is added, the array will be the same, meaning the Immutable structure will be the same, and Redux wont re-render the components.

If those are indeed plain arrays, then it should be treated as a bug and an issue needs to be open to fix this.